sdk/core/Azure.Core.TestFramework/src/Instrumentation/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -260,6 +261,58 @@ internal static async ValueTask<T> ValidateDiagnosticScope<T>(Func<ValueTask<(T
             return result;
         }
 
+        private static void CheckAttributes(Activity activity, bool strict)
+        {
+            foreach (var tag in activity.TagObjects)
+            {
+                if (tag.Key == "kind" || tag.Key.StartsWith("otel.") || tag.Key == "requestId" || tag.Key == "serviceRequestId")
+                {
+                    // TODO: these are populated on DiagnosticSource path. We should rewrite this to ActivitySource and remove those.
+                    continue;
+                }
+
+                if (!Regex.IsMatch(tag.Key, @"^[a-z\._]+$"))
+                {
+                    throw new InvalidOperationException("Attribute name can only have lowercase letters, dot (`.`), and underscore (`_`). " +
+                        "Use dot to separate namespaces and underscore to separate words (e.g. http.request.status_code). " + $"Attribute name: {tag.Key}");
+                }
+
+                int dot = tag.Key.IndexOf('.');
+                if (dot == -1)
+                {
+                    throw new InvalidOperationException("Attribute names must be namespaced. Use OpenTelemetry attributes whenever possible - https://github.com/open-telemetry/semantic-conventions/blob/main/docs/README.md. " +
+                        "Custom Azure-specific attributes must start with `az.` and should have library-specific namespaces (e.g. `az.digital_twin.twin_id`). " + $"Attribute name: {tag.Key}");
+                }
+
+                string ns = tag.Key.Substring(0, dot);
+                if ("az" != ns && "messaging" != ns && "http" != ns && "error" != ns && "url" != ns)
+                {
+                    throw new InvalidOperationException("Unknown attribute namespace. Use OpenTelemetry attributes whenever possible - https://github.com/open-telemetry/semantic-conventions/blob/main/docs/README.md." +
+                        "Custom Azure-specific attributes must start with `az.` and should have library-specific namespaces (e.g. `az.digital_twin.twin_id`). " + $"Attribute name: {tag.Key}");
+                }
+
+                string tagValueStr = tag.Value?.ToString();
+                if (string.IsNullOrEmpty(tagValueStr) || (tag.Key.StartsWith("az.") && tagValueStr.Length > 256))
+                {
+                    throw new InvalidOperationException("Attribute values must not be null, empty, or too long. "
```

Comment:
If the length limit is exactly 256, should we express that in the error?
```suggestion
                    throw new InvalidOperationException("Attribute values must not be null, empty, or longer than 256 characters. "
```